core/src/parse_sql.rs

```diff
@@ -18,124 +18,55 @@ pub fn parse<Sql: AsRef<str>>(sql: Sql) -> Result<Vec<SqlStatement>> {
-    let tokens = Tokenizer::new(&DIALECT, sql_order_by_expr.as_ref())
-        .tokenize()
-        .map_err(|e| Error::Parser(format!("{:#?}", e)))?;
-
-    Parser::new(tokens, &DIALECT)
-        .parse_order_by_expr()
-        .map_err(|e| Error::Parser(format!("{:#?}", e)))
-}
-
-pub fn parse_order_by_exprs<Sql: AsRef<str>>(
-    sql_orderby_exprs: Sql,
-) -> Result<Vec<SqlOrderByExpr>> {
-    let tokens = Tokenizer::new(&DIALECT, sql_orderby_exprs.as_ref())
-        .tokenize()
-        .map_err(|e| Error::Parser(format!("{:#?}", e)))?;
-
-    Parser::new(tokens, &DIALECT)
-        .parse_comma_separated(Parser::parse_order_by_expr)
-        .map_err(|e| Error::Parser(format!("{:#?}", e)))
-}
-
-pub fn parse_column_def<Sql: AsRef<str>>(sql_column_def: Sql) -> Result<SqlColumnDef> {
-    let tokens = Tokenizer::new(&DIALECT, sql_column_def.as_ref())
-        .tokenize()
-        .map_err(|e| Error::Parser(format!("{:#?}", e)))?;
-
-    Parser::new(tokens, &DIALECT)
-        .parse_column_def()
-        .map_err(|e| Error::Parser(format!("{:#?}", e)))
-}
-
-pub fn parse_data_type<Sql: AsRef<str>>(sql_data_type: Sql) -> Result<SqlDataType> {
-    let tokens = Tokenizer::new(&DIALECT, sql_data_type.as_ref())
-        .tokenize()
-        .map_err(|e| Error::Parser(format!("{:#?}", e)))?;
-
-    Parser::new(tokens, &DIALECT)
-        .parse_data_type()
-        .map_err(|e| Error::Parser(format!("{:#?}", e)))
-}
-
-pub fn parse_sql_assignment<Sql: AsRef<str>>(sql_assignment: Sql) -> Result<SqlAssignment> {
-    let tokens = Tokenizer::new(&DIALECT, sql_assignment.as_ref())
-        .tokenize()
-        .map_err(|e| Error::Parser(format!("{:#?}", e)))?;
-
-    Parser::new(tokens, &DIALECT)
-        .parse_assignment()
-        .map_err(|e| Error::Parser(format!("{:#?}", e)))
-}
-
-pub fn parse_identifiers<Sql: AsRef<str>>(sql_identifiers: Sql) -> Result<Vec<SqlIdent>> {
-    let tokens = Tokenizer::new(&DIALECT, sql_identifiers.as_ref())
-        .tokenize()
-        .map_err(|e| Error::Parser(format!("{:#?}", e)))?;
-
-    Parser::new(tokens, &DIALECT)
-        .parse_identifiers()
-        .map_err(|e| Error::Parser(format!("{:#?}", e)))
-}
+parse_!(parse_query, parse_query, SqlQuery);
+parse_!(parse_expr, parse_expr, SqlExpr);
+parse_!(
+    parse_comma_separated_exprs,
+    parse_comma_separated,
+    parse_expr,
+    Vec<SqlExpr>
+);
+parse_!(parse_select_item, parse_select_item, SqlSelectItem);
+parse_!(
+    parse_select_items,
+    parse_comma_separated,
+    parse_select_item,
+    Vec<SqlSelectItem>
+);
+parse_!(parse_interval, parse_interval, SqlExpr);
+parse_!(parse_order_by_expr, parse_order_by_expr, SqlOrderByExpr);
+parse_!(
+    parse_order_by_exprs,
+    parse_comma_separated,
+    parse_order_by_expr,
+    Vec<SqlOrderByExpr>
+);
+parse_!(parse_column_def, parse_column_def, SqlColumnDef);
+parse_!(parse_data_type, parse_data_type, SqlDataType);
+parse_!(parse_sql_assignment, parse_assignment, SqlAssignment);
```

Comment:
In this case, it seems that there is no need to enter sql in the middle, so I think you can delete it.